command/agent/command.go

```diff
@@ -135,6 +136,12 @@ func (c *Command) readConfig() *Config {
 			return nil
 		}
 
+		// The user asked us to load some config here but we didn't find any,
+		// so we'll complain but continue.
+		if current == nil || reflect.DeepEqual(current, &Config{}) {
```

Comment:
A consequence of changing this to an error is we cannot support delian-style `service.d` configuration directories.

In other words, you have to rewrite your service file to change configs. I think this makes more sense as a warning which is less stringent about this type of thing.